### PR TITLE
Aw/72 hour after ip follow up report 181797631

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: drivers/health_ip_followup_report/app/models/health_ip_followup_report/followups_report.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    def initialize(start_date:, end_date:)
+      @range = (start_date..end_date)
+    end
+
+    def qas_by_month
+      @qas_by_month ||= Health::QualifyingActivity.
+        submittable.
+        in_range(@range).
+        where(activity: :discharge_follow_up).
+        group(Arel.sql("DATE_TRUNC('MONTH', date_of_activity)")).
```
Comment (eanders) [inline drivers/health_ip_followup_report/app/models/health_ip_followup_report/followups_report.rb:18]: You are limiting to a 3 month increment in the controller, but potentially, in the future, that could be expanded (or maybe someone can pick more than a year in the UI.  This only groups by month, so potentially you'll group 1/2020 with 1/2021 and 1/2022.  Might want to group by year and month.

Comment (eanders) [reply drivers/health_ip_followup_report/app/models/health_ip_followup_report/followups_report.rb:18]: My mistake, date_trunc retains year, I was thinking of date_part.

File: drivers/health_ip_followup_report/app/models/health_ip_followup_report/followups_report.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    def dates
+      (qas_by_month.keys + visits_by_month.keys).uniq
+    end
+
+    def infill(data)
+      dates.map do |date|
+        data[date] || 0
+      end
+    end
```
Comment (eanders) [inline drivers/health_ip_followup_report/app/models/health_ip_followup_report/followups_report.rb:40]: That's tricky!

**Review by eanders — APPROVED** [review]

**Comment by awisema** [comment]
BHCHP wanted an non-admin to be able to see this report -- because it is just aggregate data, I related the permissions required to just being able to view aggregate health data.

**Pull request merged**